core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -144,10 +144,12 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
    * be deleted on JVM exit when using the external shuffle service.
    */
   private def createLocalDirs(conf: SparkConf): Array[File] = {
+    if (deleteFilesOnStop) LocalDirectoryCleanupService.clean()
```

Comment:
Since this flag can change after restart, I think its better to invoke clean() without the check for deleteFilesOnStop